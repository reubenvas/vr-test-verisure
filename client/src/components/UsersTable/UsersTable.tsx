import React, { useEffect, useState } from 'react';
import styles from './styles.scss'; // eslint-disable-line @typescript-eslint/no-unused-vars
import UserTableRow from '../UserTableRow/UserTableRow';
import { getUsers } from '../../api/userApi';

type users = {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
}[] | null;

const UsersTable = (): React.ReactElement => {
    const [users, setUsers] = useState<users>([]);

    const fetchUsers = async (): Promise<void> => {
        try {
            const response = await getUsers();
            setUsers(response);
        } catch (err) {
            console.log(err); // eslint-disable-line no-console
        }
    };

    useEffect(() => {
        fetchUsers();
    }, []);

    console.log(process.env.TEST_GREETING);

    return (
        <table>
            <thead>
                <tr>
                    <th>&nbsp;</th>
                    <th>Id</th>
                    <th>First Name</th>
                    <th>Last Name</th>
                    <th>Email</th>
                </tr>
            </thead>
            <tbody id="users">
                {users && users.map((user) => (
                    <UserTableRow
                        key={user.id}
                        id={user.id}
                        firstName={user.firstName}
                        lastName={user.lastName}
                        email={user.email}
                    />
                ))}
            </tbody>
        </table>
    );
};

export default UsersTable;
