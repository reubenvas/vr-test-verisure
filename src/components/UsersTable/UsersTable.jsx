import React, { useEffect, useState } from 'react';
import UserTableRow from '../UserTableRow/UserTableRow';
import { getUsers } from '../../api/userApi';

const UsersTable = () => {
    const [users, setUsers] = useState([]);

    const fetchUsers = async () => {
        const response = await getUsers();
        setUsers(response);
    };

    useEffect(() => {
        fetchUsers();
    }, []);

    return (
        <table>
            <thead>
                <tr>
                    <th>&nbsp;</th>
                    <th>Id</th>
                    <th>First Name</th>
                    <th>Last Name</th>
                    <th>Email</th>
                </tr>
            </thead>
            <tbody id="users">
                {users.map((user) => (
                    <UserTableRow
                        key={user.id}
                        id={user.id}
                        firstName={user.firstName}
                        lastName={user.lastName}
                        email={user.email}
                    />
                ))}
            </tbody>
        </table>
    );
};

export default UsersTable;
