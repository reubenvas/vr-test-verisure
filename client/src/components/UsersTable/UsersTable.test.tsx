import React from 'react';
import renderer from 'react-test-renderer';
import UsersTable from './UsersTable';

describe('DarkButton component', () => {
    test('snapshot renders', async () => { // doesn't snapshot after the component has fetched data
        const component = renderer.create(<UsersTable />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
