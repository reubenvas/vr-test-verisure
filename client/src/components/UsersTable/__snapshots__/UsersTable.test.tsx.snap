// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Â 
      </th>
      <th>
        Id
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody
    id="users"
  />
</table>
`;
